woocommerce/woocommerce
# Adding context property to tasklist_view and task_click events

### All Submissions:






### Changes proposed in this Pull Request:

Adding the `context` property to the `wcadmin_tasklist_view` and `wcadmin_tasklist_click` tracks events, indicating if the tasklist was in the sidebar or homescreen.

I also took the liberty of fixing a couple bugs I came across involving the ActionPanel being displayed incorrectly when the reminder bar was visible, and the `wcadmin_tasklist_view` event being fired for lists that were not visible since they had no tasks.

Closes #33284  .

### How to test the changes in this Pull Request:

1. Checkout branch on fresh site, no need to complete OBW.
2. Go to a screen where the reminder bar is present (ie Customers) and click "Finish Setup." Ensure that the top of the panel aligns correctly with the header. 
3. Enable tracks logging in your browser console with `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
4. When you load the Homescreen, ensure that the `wcadmin_extended_tasklist_view` is NOT fired since it's not visible.
5. View Homescreen, and look for the `wcadmin_tasklist_view` event logged in your console. Ensure it has the `context` property with a value of `homescreen`.
6. Click on one of the tasks, and look for the `wcadmin_tasklist_click` event in your console. Ensure it also has the `context` property with a value of `homescreen`.
7. Now navigate away from the homescreen to another page with the "Finish Setup" button in the activity panel. (ie Customers or Orders)
8. Click the "Finish Setup" button to bring out the tasks sidebar. 
9. You should now see the `wcadmin_tasklist_view` event fire again. Ensure that it has a `context` property of `sidebar`.
10. Click on a task, and look for the `wcadmin_tasklist_click` event again. Ensure that it also has a `context` property of `sidebar`.
11. Using the WCA Test Helper plugin, enable the `woocommerce_tasklist_setup_experiment_1_*` experiment and do steps 4-8 again. 
12. Using the WCA Test Helper plugin, enable the `woocommerce_tasklist_setup_experiment_2_*` experiment and do steps 4-8 again. 

### Other information:




### FOR PR REVIEWER ONLY:
